Readme and Release notes changes for rc2 (#87)
notaryproject/roadmap: RELEASENOTES/v2.0.0.rc-2.MD

> # Notary v2 Release Candidate 2 

Reviewer: We should remove all references to v2 in the documents. Please use Notary Project instead. I will not comment on every line including the reference - we should do a bulk update.

- [Notation Library 1](https://github.com/notaryproject/notation-core-go) ***[(notation-core-go-v1.0.0-rc.2)](https://github.com/notaryproject/notation-core-go/releases/tag/v1.0.0-rc.2)***   (Updated in this release)
- [Notation Library 2](https://github.com/notaryproject/notation-go) ***[(notation-go-v1.0.0-rc.3)](https://github.com/notaryproject/notation-go/releases/tag/v1.0.0-rc.3)***   (Updated in this release)
- [Notation CLI](https://github.com/notaryproject/notation)   ***([notation-v1.0.0-rc.2)](https://github.com/notaryproject/notation/releases/tag/v1.0.0-rc.2)***   (Updated in this release) 
- [Notary v2 Specs](https://github.com/notaryproject/notaryproject) ***[(notaryproject-v1.0.0-rc.2)](https://github.com/notaryproject/notaryproject/releases/tag/v1.0.0-rc.2)*** (Updated in this release)

## Goal of the release
- Notary v2 specifications

>   - Updates to support signing and verification using the fallback method for referrers, as specified in the 1.1 version release candidate of the OCI Distribution specification. With this method, Notary v2 now supports a reference types solution but within the bounds of OCI 1.0 Image and Distribution specs 

Reviewer 2: ```suggestion
  - Support signing and verification using the fallback method to improve backward compatibility with OCI v1.0 compliant registry and comply with OCI distribution-spec and image-spec v1.0. With this method, Notary v2 now supports a reference types solution that allows you to enable supply chain artifact reference.
```

- Update "Notation" (CLI client) implementation to match the latest specifications. With this new version of Notation client, users can

>   - Choose the method to manage signatures inside their registries - image manifest or artifact manifest.

Reviewer 2: ```suggestion
  - Able to choose a specific OCI manifest type (OCI image manifest or OCI artifact manifest) to store the signature, alongside with OCI artifact in a registry.
```

>   - Provide optional meta-data at time of signing as key-value pairs.

Reviewer 2: ```suggestion
  - Able to add key-value pairs to the signature payload when signing an OCI artifact
```

## List of feature/capabilities added with this release
For detailed notes on each of the individual repositories (NotaryProject, Notation (for reference CLI implementation) and Notation libraries) users can refer to the individual release notes. A summary view is provided in this document.

### Notation CLI Client
Notation is a standalone CLI client based on Notary v2 specs. It supports signing, verifying, and storing signatures in registries supporting signatures as references to images.

In this release of Notation command, the following CLI commands are supported. 
  - **sign**        Sign OCI artifacts
  - **verify**      Verify OCI artifacts
  - **certificate** Manage certificates in trust store for verifying 
  - **key**         Manage keys used for signing
  - **list**        List signatures of the signed artifact
  - **login**       Login to registry
  - **logout**      Log out from the logged in registries
  - **plugin**      Manage plugins  
  - **version**     Show the notation version information
  - **help**        Help about any command

*One new command were added in this RC-2 release*

>   - **inspect**     Inspect all signatures associated with the signed artifact 

Reviewer 2: @vaninrao10 @iamsamirzon @yizha1  Do you think it's proper to mention a newly-added flag `--user-metadata` in `notation sign` here?

Author: This is only commands not the flags in the release notes. I would say it is not needed. Will let others also add their opinion

  
As part of this release, users have to create a trust policy configuration file before verifying arifacts. 

> To start using Notation Refer: https://notaryproject.dev/docs/quickstart/ 

Reviewer 2: ```suggestion
To start using Notation, please refer to the [Notation quick start](https://notaryproject.dev/docs/quickstart/).
```

### Notation Library 

> Written in the GO language the Library is divided up into two separate repos - [Notation-go](https://github.com/notaryproject/notation-go) and [Notation-core-go](https://github.com/notaryproject/notation-core-go). Updated Notation-go for

Reviewer 2: Updated Notation-go with the following enhancements:

> Written in the GO language the Library is divided up into two separate repos - [Notation-go](https://github.com/notaryproject/notation-go) and [Notation-core-go](https://github.com/notaryproject/notation-core-go). Updated Notation-go for

Reviewer 2: ```suggestion
Written in the Golang language, the library is divided up into two separate repositories, [Notation-go](https://github.com/notaryproject/notation-go) and [Notation-core-go](https://github.com/notaryproject/notation-core-go). Notation-go was updated with the following enhancements:
```

> - Adding optional meta-data at time of signing images

Reviewer 2: ```suggestion
- Added optional metadata at the time of signing images
```

- Added check for plugin versioning and compatability

## List of fixed issues
Refer individual release notes.

## Roadmap Guidance

> *This is not an exhaustive list. Users should refer to the Notary v2 roadmap at https://github.com/notaryproject/roadmap#readme for seeing the plan for subsequent releases and refer to individual subprojects (repositories) releases notes for details*

Reviewer 2: ```suggestion
The major enhancements planned for Notation v1.0.0-RC.3 include but are not limited to the following list. Users should refer to the Notary v2 roadmap at https://github.com/notaryproject/roadmap#readme for seeing the plan for subsequent releases and refer to individual subprojects (repositories) releases notes for details*
```

1. Revocation support via Notation client will come in a future release. Spec update for revocation support is planned for the next release candidate
2. CLI for managing trust policy is planned for the next release candidate
3. Sign/verify offline artifacts

> 4. UX improvement for notation plugin

Reviewer 2: 5. Simplify and improve Notation CLI installation by adding the installer for each popular platform/OS (See   https://github.com/notaryproject/roadmap/issues/76)

### Backwards Compatibility
- This version of Notation client is compatible with RC-1, the prior releases of Notation client.

> - This version of Notation client is meant for registries compliant to both the OCI [1.0.2 image spec](https://github.com/opencontainers/image-spec/releases/tag/v1.0.2) or [1.1.0-rc2 image spec](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0-rc2). The version is also compliant to the OCI [1.1.0-rc1 distribution spec](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0-rc1) 

Reviewer 2: ```suggestion
- This version of the Notation client is meant for registries compliant with both the OCI [1.0.2 image spec](https://github.com/opencontainers/image-spec/releases/tag/v1.0.2) and [1.1.0-rc2 image spec](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0-rc2). The version is also compliant with the OCI [1.1.0-rc1 distribution spec](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0-rc1) 
```

- [Notation CLI](https://github.com/notaryproject/notation)   ***([notation-v1.0.0-rc.2)](https://github.com/notaryproject/notation/releases/tag/v1.0.0-rc.2)***   (Updated in this release) 
- [Notary v2 Specs](https://github.com/notaryproject/notaryproject) ***[(notaryproject-v1.0.0-rc.2)](https://github.com/notaryproject/notaryproject/releases/tag/v1.0.0-rc.2)*** (Updated in this release)

## Goal of the release
- Notary v2 specifications
    - Updates to support signing and verification using the fallback method for referrers, as specified in the 1.1 version release candidate of the OCI Distribution specification. With this method, Notary v2 now supports a reference types solution but within the bounds of OCI 1.0 Image and Distribution specs 
 - Update "Notation" (CLI client) implementation to match the latest specifications. With this new version of Notation client, users can
    - Choose the method to manage signatures inside their registries - image manifest or artifact manifest.
    - Provide optional meta-data at time of signing as key-value pairs.
    - Use a new "inspect" command to see relevant details encoded in all signtures associated with an image.

>     - Use a new "completion" command to generate the corresponding completion script for their shells. 

Reviewer 3: Is this new command? I think this is introduced by the CLI framework cobra @patrickzheng200  Could you confirm it?

For detailed notes on each of the individual repositories (NotaryProject, Notation (for reference CLI implementation) and Notation libraries) users can refer to the individual release notes. A summary view is provided in this document.

### Notation CLI Client
Notation is a standalone CLI client based on Notary v2 specs. It supports signing, verifying, and storing signatures in registries supporting signatures as references to images.

In this release of Notation command, the following CLI commands are supported. 
  - **sign**        Sign OCI artifacts
  - **verify**      Verify OCI artifacts
  - **certificate** Manage certificates in trust store for verifying 
  - **key**         Manage keys used for signing
  - **list**        List signatures of the signed artifact
  - **login**       Login to registry
  - **logout**      Log out from the logged in registries
  - **plugin**      Manage plugins  
  - **version**     Show the notation version information
  - **help**        Help about any command
  *The following two new commands were added in this RC-2 release*
  - **inspect**     Inspect all signatures associated with the signed artifact 

>   - **completion**  Generate the autocompletion script for the specified shell

Reviewer 3: If this is from the CLI framework, I suggest not mentioning it.

Author: It from CLI framework, and it shows up in the Notation -- help. Hence, we decided to add a line in there.

As part of this release, users have to create a trust policy configuration file before verifying arifacts. 

To start using Notation Refer: https://notaryproject.dev/docs/quickstart/ 

### Notation Library 
Written in the GO language the Library is divided up into two separate repos - [Notation-go](https://github.com/notaryproject/notation-go) and [Notation-core-go](https://github.com/notaryproject/notation-core-go). Updated Notation-go for
- Adding optional meta-data at time of signing images

> - Added  check for plugin versioning and compatability

Reviewer 3: nit:
remove additional space

## List of fixed issues
Refer individual release notes.

## Roadmap Guidance
*This is not an exhaustive list. Users should refer to the Notary v2 roadmap at https://github.com/notaryproject/roadmap#readme for seeing the plan for subsequent releases and refer to individual subprojects (repositories) releases notes for details*

1. Revocation support via Notation client will come in a future release. Spec update for revocation support is planned for the next release candidate

> 2. CLI specs for managing trust policy is planned for the next release candidate

Reviewer 3: I suggest removing specs because implementation will also be done at least for phase-1

> 

Reviewer 3: Could you add the follows based on our discussions?
- Sign/verify offline artifacts
- UX improvement for notation plugin

### Backwards Compatibility

> - This version of Notation client is  compatible with RC-1, the prior releases of Notation client.

Reviewer 3: nit:
remove additional space.
